scripts/plot_scalar_resources.py

```diff
@@ -33,33 +35,76 @@ def load_scalars(path):
     # Load scalar data
     scalars = load_scalars(resources)
 
-    def unstack_scalars(df):
-        _df = df.copy()
-        _df = _df.unstack("var_name")
-
-        return _df
-
-    scalars.set_index(["region", "carrier", "tech", "var_name"], inplace=True)
-    scalars = scalars["var_value"]
-
-    # There are duplicate entries for those capacities where we distinguish chp=yes/no
-    # TODO: Rethink this
-    scalars = scalars.loc[scalars.index.duplicated()]
-
-    # unstack
-    scalars_prepared = unstack_scalars(scalars)
-
-    # filter
-    scalars_prepared = scalars_prepared["capacity_net_el"]
-
-    # Set numeric datatype
-    scalars_prepared = scalars_prepared.astype(float)
-
-    # plot
-    fig, ax = plt.subplots()
-
-    scalars_prepared.plot.bar(ax=ax)
-
-    plt.tight_layout()
-
-    plt.savefig(target)
+    # GENERAL
+    unit_dict = {"capacity_net_el": "W"}
+    carrier_dict = {
+        "biomass": "Biomass",
+        "ch4": "CH4",
+        "hard coal": "Hard coal",
+        "oil": "Oil",
+        "lignite": "Lignite",
+        "other": "Other",
+    }
+
+    # User input
+    var_name = "capacity_net_el"
+    ylabel = "Leistung [MW]"
+
+    def prepare_conv_pp_scalars(var_name, carrier_dict=carrier_dict):
```

Comment:
I would suggest move the definitions and functions to the to, above `if __name__ == '__main__`:.